ooniapi/services/oonifindings/src/oonifindings/routers/v1.py

```diff
@@ -191,212 +208,203 @@ def get_oonifinding_by_id(
-    text, event_type, published, CCs, ASNs, domains, tags, links,
-    test_names, short_description, email_address, create_time)
-    VALUES
-    """
-    r = insert_click(db, query, [incident_dict])
-    optimize_table(db, tblname="incidents")
-    
+    db_oonifinding = models.OONIFinding(
+        incident_id=incident_id,
+        create_time=now,
+        update_time=now,
+        start_time=create_request.start_time,
+        end_time=create_request.end_time,
+        creator_account_id=account_id,
+        title=create_request.title,
+        short_description=create_request.short_description,
+        text=create_request.text,
+        reported_by=create_request.reported_by,
+        email_address=create_request.email_address,
+        event_type=create_request.event_type,
+        published=int(create_request.published),
+        CCs=create_request.CCs,
+        ASNs=create_request.ASNs,
+        domains=create_request.domains,
+        tags=create_request.tags,
+        links=create_request.links,
+        test_names=create_request.test_names,
+    )
+
+    db.add(db_oonifinding)
+    db.commit()
+
     setnocacheresponse(response)
-    return OONIFindingsUpdateResponse(r=r, id=incident_id)
+    return OONIFindingsUpdateResponse(r=1, id=incident_id)
 
 
 @router.post(
     "/v1/incidents/update",
-    dependencies=[Depends(role_required(["admin", "user"]))],
     tags=["oonifindings"],
-    response_model=OONIFindingsUpdateResponse    
+    response_model=OONIFindingsUpdateResponse
 )
 def update_oonifinding(
     update_request: OONIFindingCreateUpdate,
     response: Response,
-    authorization: str = Header("authorization"),
-    db=Depends(get_clickhouse_session),
+    db=Depends(get_postgresql_session),
     token=Depends(role_required(["admin", "user"])),
-    settings=Depends(get_settings)
 ):
     """
     Update an incident
-    """ 
-    incident_id = update_request.incident_id 
-    if token["role"] != "admin":
-        verify_user(
-            db,
-            authorization=authorization,
-            jwt_encryption_key=settings.jwt_encryption_key,
-            incident_id=incident_id,
-            email_address=update_request.email_address,
-            key=settings.account_id_hashing_key,
-        )
-        
-        if update_request.published is True:
-            raise HTTPException(status_code=400, detail="Not enough permissions to publish")
+    """
+    incident_id = update_request.incident_id
+    account_id = token["account_id"]
 
-    update_request.creator_account_id = get_account_id_or_raise(
-        authorization, jwt_encryption_key=settings.jwt_encryption_key
-    ) 
-    incident_dict = prepare_incident_dict(update_request)
+    q = db.query(models.OONIFinding).filter(
+        models.OONIFinding.incident_id == incident_id
+    )
+    if token["role"] == "user":
+        q = q.filter(models.OONIFinding.creator_account_id == account_id)
```

Comment:
I wonder if you can just do all of this in a single query, where you get one row and then apply the authentication checks once you have the row depending on the token role.

Reply:
Thanks for the suggestion. You are right. I made some changes here: https://github.com/ooni/backend/pull/862/commits/f4c97afffc0abe206abbb457a6afddfe61684f78. In most cases, I have tried to assemble the query logic into a single query.